tests/unit/adaptors/graph/test_layers.py

```diff
@@ -958,3 +958,79 @@ def _pairwise(iterable):
     a, b = itertools.tee(iterable)
     next(b, None)
     return zip(a, b)
+
+
+class TestClosedLayers:
```

Comment:
I think we're missing tests for layers with multiple closed layers, right?

Reply:
Maybe. I'm unsure exactly what we want to test here... I've added another unit test case for `generate_module_permutations` for two closed layers - does this cover what you were hoping to test?